src/components/KYCWall/BaseKYCWall.tsx

```diff
@@ -181,11 +118,19 @@ function KYCWall({
                     return;
                 }
 
-                const clickedElementLocation = getClickedTargetLocation(targetElement as HTMLDivElement);
-                const position = getAnchorPosition(clickedElementLocation);
-
-                setPositionAddPaymentMenu(position);
-                setShouldShowAddPaymentMenu(true);
+                switch (iouPaymentType) {
+                    case CONST.PAYMENT_METHODS.PERSONAL_BANK_ACCOUNT:
+                        selectPaymentMethod(CONST.PAYMENT_METHODS.PERSONAL_BANK_ACCOUNT);
```

Comment:
Shouldn't this show up only when `shouldShowPersonalBankAccountOption` is `true`?